sycl/plugins/unified_runtime/ur/adapters/hip/device.cpp

```diff
@@ -857,6 +866,8 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGet(ur_platform_handle_t hPlatform,
 
   try {
     UR_ASSERT(pNumDevices || phDevices, UR_RESULT_ERROR_INVALID_VALUE);
+    UR_ASSERT((NumEntries == 0 && !phDevices) || (NumEntries > 0 && phDevices),
```

Comment:
should this be caught by the UR validation layer, instead of having to add it to the adapter?

Reply:
Thanks 👍  I've removed this check